internal/controller/controllers/agent_controller.go

```diff
@@ -905,6 +915,50 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 	return nil
 }
 
+func (r *AgentReconciler) getIgnitionToken(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent) (string, error) {
+	ignitionEndpointTokenReference := agent.Spec.IgnitionEndpointTokenReference
+
+	if ignitionEndpointTokenReference == nil {
+		log.Info("No IgnitionEndpointTokenReference set")
+		return "", errors.New("No IgnitionEndpointTokenReference set")
```

Comment:
Unsure this should be an error.
@avishayt shouldn't we sync a token removal?

Reply:
Agree.  Also, this will never happen because we call this function only if the reference is not nil.  So we need to change that too.

Reply:
.